services/iplant-groups/src/iplant_groups/clients/grouper.clj

```diff
@@ -611,25 +642,80 @@
         first)))
 
 (defn assign-role-permission
-  [username attribute-def-name-id role-id allowed? action-names]
+  [username attribute-def-name role-name allowed? action-names]
   (assign-remove-permission
     (format-role-permission-assign-request
-      username attribute-def-name-id role-id allowed? action-names)))
+      username attribute-def-name role-name allowed? action-names)))
 
 (defn remove-role-permission
-  [username attribute-def-name-id role-id action-names]
+  [username attribute-def-name role-name action-names]
   (assign-remove-permission
     (format-role-permission-remove-request
-      username attribute-def-name-id role-id action-names)))
+      username attribute-def-name role-name action-names)))
 
 (defn assign-membership-permission
-  [username attribute-def-name-id role-id subject-id allowed? action-names]
+  [username attribute-def-name role-name subject-id allowed? action-names]
   (assign-remove-permission
     (format-membership-permission-assign-request
-      username attribute-def-name-id role-id subject-id allowed? action-names)))
+      username attribute-def-name role-name subject-id allowed? action-names)))
 
 (defn remove-membership-permission
-  [username attribute-def-name-id role-id subject-id action-names]
+  [username attribute-def-name role-name subject-id action-names]
   (assign-remove-permission
     (format-membership-permission-remove-request
-      username attribute-def-name-id role-id subject-id action-names)))
+      username attribute-def-name role-name subject-id action-names)))
+
+(defn- format-all-permission-search-request
+  [username attribute-def-name]
+  {:WsRestGetPermissionAssignmentsRequest
+   (remove-vals nil? {:actAsSubjectLookup (act-as-subject-lookup username)
+                      :wsAttributeDefNameLookups [{:name attribute-def-name}]})})
+
+(defn- get-permission-assign-ids
+  [username attribute-def-name]
+  (->> (format-all-permission-search-request username attribute-def-name)
+       (permission-assignment-search*)
+       (distinct-by :attributeAssignId)
+       (group-by :permissionType)
+       (map-kv (fn [k v] [k (mapv :attributeAssignId v)]))))
+
+(defn- format-permission-assign-id-removal-request
+  [username permission-type ids]
+  {:WsRestAssignPermissionsRequest
+   {:permissionAssignOperation "remove_permission"
+    :actAsSubjectLookup (act-as-subject-lookup username)
+    :permissionType permission-type
+    :wsAttributeAssignLookups (mapv (partial hash-map :uuid) ids)}})
+
+(defn- remove-permission-assign-ids
+  [username permission-type ids]
+  (->> (format-permission-assign-id-removal-request username permission-type ids)
+       (assign-remove-permission)))
+
+(defn- remove-existing-permissions
+  [username attribute-def-name]
+  (->> (get-permission-assign-ids username attribute-def-name)
+       (map (fn [[permission-type ids]] (remove-permission-assign-ids username permission-type ids)))
+       dorun))
```

Comment:
This is where the calls to remove existing permissions are actually made. The output from `get-permission-assign-ids` is designed specifically to allow us to iterate over it while removing permissions.